src/Fcf/Data/List.hs

```diff
@@ -311,19 +444,235 @@ type instance Eval (DropWhile p (x ': xs)) =
       (Pure (x ': xs)))
 
 
--- Helper for Reverse. This corresponds to rev in the data list lib.
-data Rev :: [a] -> [a] -> Exp [a]
-type instance Eval (Rev '[]       ys) = ys
-type instance Eval (Rev (x ': xs) ys) = Eval (Rev xs (x ': ys))
+--------------------------------------------------------------------------------
+
+-- | 'Span', applied to a predicate @p@ and a type-level list @xs@, returns a 
+-- type-level tuple where
+-- first element is longest prefix (possibly empty) of @xs@ of elements that
+-- satisfy @p@ and second element is the remainder of the list:
+--
+-- === __Example__
+-- 
+-- >>> :kind! Eval (Span (Flip (<) 3) '[1,2,3,4,1,2,3,4])
+-- Eval (Span (Flip (<) 3) '[1,2,3,4,1,2,3,4]) :: ([Nat], [Nat])
+-- = '( '[1, 2], '[3, 4, 1, 2, 3, 4])
+--
+-- >>> :kind! Eval (Span (Flip (<) 9) '[1,2,3])
+-- Eval (Span (Flip (<) 9) '[1,2,3]) :: ([Nat], [Nat])
+-- = '( '[1, 2, 3], '[])
+--
+-- >>> :kind! Eval (Span (Flip (<) 0) '[1,2,3])
+-- Eval (Span (Flip (<) 0) '[1,2,3]) :: ([Nat], [Nat])
+-- = '( '[], '[1, 2, 3])
+data Span :: (a -> Exp Bool) -> [a] -> Exp ([a],[a])
+type instance Eval (Span p lst) = '( Eval (TakeWhile p lst), Eval (DropWhile p lst))
 
 
--- | Type-level list reverse.
+-- | 'Break', applied to a predicate @p@ and a type-level list @xs@, returns a 
+-- type-level tuple where
+-- first element is longest prefix (possibly empty) of @xs@ of elements that
```

Comment:
**the** first element is **the** longest prefix

Reply:
Added 'the' in there (note, I copied the text from Data.List and it doesn't contain those).